### PR TITLE
Add version file size indicator

### PR DESCRIPTION
**Pull request by RockyTV**
Not sure if it works, I reworked my original PR (#133) to adapt to the new branch.

Closes #200

### CODE REVIEW

**Comment by HebaruSan** [comment]
Can you also update the buttons for the historical buttons under Changelog?

https://github.com/KSP-SpaceDock/SpaceDock/blob/93466ef306a9718f770ebf624a1d230213ccc3c8/templates/mod.html#L346

File: KerbalStuff/blueprints/mods.py [hunk 1]
```diff
@@ -128,7 +128,7 @@ def mod(mod_id, mod_name):
         follower_stats.append(dumb_object(f))
     json_versions = list()
     for v in mod.versions:
-        json_versions.append({ 'name': v.friendly_version, 'id': v.id })
+        json_versions.append({ 'name': v.friendly_version, 'id': v.id, 'file_size': get_version_size(os.path.join(__cfg('storage'), v.download_path) })
```
Comment (DasSkelett) [inline KerbalStuff/blueprints/mods.py]: You are missing a closing parenthesis here.

**Comment by HebaruSan** [comment]
Looks good to me. Thanks for re-doing this! Should be live on the alpha server in a couple of minutes...

**Comment by HebaruSan** [comment]
... nope, it's broke. 500 status for mod pages now.

https://alpha.spacedock.info/mod/1/Astrogator

**Comment by HebaruSan** [comment]
And now the `__cfg`:

```
Aug 01 02:24:54 sd6 gunicorn[14675]:   File "/var/www/virtual/spacedock.info/htdocs/SpaceDock/KerbalStuff/blueprints/mods.py", line 131, in mod
Aug 01 02:24:54 sd6 gunicorn[14675]:     json_versions.append({ 'name': v.friendly_version, 'id': v.id, 'file_size': get_version_size(os.path.join(__cfg('storage'), v.download_path)) })
Aug 01 02:24:54 sd6 gunicorn[14675]: NameError: name '__cfg' is not defined
```

**Comment by HebaruSan** [comment]
And another missing import:

```
Aug 01 02:30:23 sd6 gunicorn[15151]:   File "/var/www/virtual/spacedock.info/htdocs/SpaceDock/KerbalStuff/common.py", line 203, in get_version_size
Aug 01 02:30:23 sd6 gunicorn[15151]:     if not os.path.isfile(f): return None
Aug 01 02:30:23 sd6 gunicorn[15151]: NameError: name 'os' is not defined
```

**Comment by RockyTV** [comment]
Do you want me to fix these issues?

**Comment by HebaruSan** [comment]
Fixed all that&mdash;and it still doesn't work! :(

image

**Comment by RockyTV** [comment]
This is weird. The file size is calculated correctly but it's not being displayed. Printing `window.versions` on alpha returned:

```
[
  {
    "file_size": "78.71 KiB",
    "id": 1,
    "name": "v0.9.2"
  }
]
```

**Comment by HebaruSan** [comment]
Ahh, `json_versions` was the wrong variable to update. It's not used for rendering the page, it just goes into `window.versions`, which isn't used for anything. The `mod` variable is the one that is used for rendering.

**Comment by HebaruSan** [comment]
It would be really nice if we could figure out things like this before merging. @DasSkelett , does that docker stuff result in a fully functional site?

**Comment by RockyTV** [comment]
Okay, that's going to be a bit more complicated to do. Hopefully we don't have to mess around with sqlalchemy.



> It would be really nice if we could figure out things like this before merging.

Maybe if we setup Travis with a linter we can prevent these issues (import, wrong function name) from happening?

**Comment by HebaruSan** [comment]
> Maybe if we setup Travis with a linter we can prevent these issues (import, wrong function name) from happening?

Some, maybe, but that still would not have told us that the enhancement doesn't work. We'd need actual testing for that.

**Comment by HebaruSan** [comment]
I'm going to try reverting this. Not sure whether I'll be able to force push on the alpha branch, but it's worth a try.

**Comment by DasSkelett** [comment]
> It would be really nice if we could figure out things like this before merging. @DasSkelett , does that docker stuff result in a fully functional site?

It does! That's how I found the parenthesis syntax error. A nice stack trace during startup. It's as simple as `[sudo] docker-compose up`. Now there's a webpage on `localhost:5000`.
There are a placeholder game, publisher, and a user and admin account preinitialized, see the `db_initializy.py`. Some tips can be found in the `README_DOCKERFILE.md`, e.g. how to open a console inside one of the containers to check the database or stuff.

Could have tested this PR more before merge, but you managed to do that in the middle of the night for me.

**Comment by HebaruSan** [comment]
> Some tips can be found in the `README_DOCKERFILE.md`, e.g. how to open a console inside one of the containers to check the database or stuff.

That sounds excellent. Maybe we should write up a "how to develop for SpaceDock" wiki, containing a basic docker intro and an explanation of the alpha/beta branches and servers. Or even a pull request template with a checklist including "did you test this in docker".

**Comment by DasSkelett** [comment]
> That sounds excellent. Maybe we should write up a "how to develop for SpaceDock" wiki, containing a basic docker intro and an explanation of the alpha/beta branches and servers. Or even a pull request template with a checklist including "did you test this in docker".

Good idea. However I can't create or edit wiki pages right now in this repo.
Pinging @V1TA5, do you want to open up wiki editing to everyone?

**Pull request merged**